prelude/README.md

```diff
@@ -782,49 +768,169 @@ instance ordArray :: (Ord a) => Ord [a]
 ```
 
 
-#### `Bits`
+#### `Bounded`
 
 ``` purescript
-class Bits b where
-  (.&.) :: b -> b -> b
-  (.|.) :: b -> b -> b
-  (.^.) :: b -> b -> b
-  shl :: b -> Number -> b
-  shr :: b -> Number -> b
-  zshr :: b -> Number -> b
-  complement :: b -> b
+class (Ord a) <= Bounded a where
+  top :: a
+  bottom :: a
 ```
 
-The `Bits` type class identifies types which support bitwise operations.
+The `Bounded` type class represents types that are finite partially
+ordered sets.
 
-#### `bitsNumber`
+Instances should satisfy the following law in addition to the `Ord` laws:
+
+- Ordering: `bottom <= top`
+
+#### `boundedBoolean`
 
 ``` purescript
-instance bitsNumber :: Bits Number
+instance boundedBoolean :: Bounded Boolean
 ```
 
 
-#### `BoolLike`
+#### `Lattice`
 
 ``` purescript
-class BoolLike b where
-  (&&) :: b -> b -> b
-  (||) :: b -> b -> b
-  not :: b -> b
+class (Ord a) <= Lattice a where
+  join :: a -> a -> a
+  meet :: a -> a -> a
 ```
 
-The `BoolLike` type class identifies types which support Boolean operations.
+The `Lattice` type class represents types that are partially ordered
+sets with a join (`||`) and meet (`&&`) operation.
 
-`BoolLike` instances are required to satisfy the laws of a _Boolean algebra_.
+Instances should satisfy the following laws in addition to the `Ord`
+laws:
 
+- Associativity:
+  - `a || (b || c) = (a || b) || c`
+  - `a && (b && c) = (a && b) && c`
+- Commutativity:
+  - `a || b = b || a`
+  - `a && b = b && a`
+- Absorption:
+  - `a || (a && b) = a`
+  - `a && (a || b) = a`
+- Idempotent:
+  - `a || a = a`
+  - `a && a = a`
 
-#### `boolLikeBoolean`
+#### `latticeBoolean`
 
 ``` purescript
-instance boolLikeBoolean :: BoolLike Boolean
+instance latticeBoolean :: Lattice Boolean
 ```
 
 
+#### `BoundedLattice`
+
+``` purescript
+class (Bounded a, Lattice a) <= BoundedLattice a where
+```
+
+The `BoundedLattice` type class represents types that are finite
+lattices.
+
+Instances should satisfy the following law in addition to the `Lattice`
+and `Bounded` laws:
+
+- Identity:
+  - `a || bottom = a`
+  - `a && top = a`
+
+#### `boundedLatticeBoolean`
+
+``` purescript
+instance boundedLatticeBoolean :: BoundedLattice Boolean
+```
+
+
+#### `ComplementedLattice`
+
+``` purescript
+class (BoundedLattice a) <= ComplementedLattice a where
+  not :: a -> a
+```
+
+The `ComplementedLattice` type class represents types that are lattices
+where every member is also complemented.
+
+Instances should satisfy the following law in addition to the
+`BoundedLattice` laws:
+
+- Complemented:
+  - For every `a` in interval `[c, d]` there should be a `b` such that `a || b = top`
```

Comment:
`[c, d]` is just "any interval in the poset", I guess since `c` and `d` aren't used here (unlike the relatively complemented law) we could just say it that way instead?